Framework/Indexing/inc/MantidIndexing/IndexInfo.h

```diff
@@ -91,6 +100,9 @@ class MANTID_INDEXING_DLL IndexInfo {
 
   Kernel::cow_ptr<std::vector<specnum_t>> m_spectrumNumbers;
   Kernel::cow_ptr<std::vector<std::vector<detid_t>>> m_detectorIDs;
+
+  Kernel::cow_ptr<std::vector<Beamline::SpectrumDefinition>>
```

Comment:
As soon as a `SpectrumDefinition` is changed sharing will be and must be broken. Currently this will mainly happen when using, e.g., `ISpectrum::setDetectorID()`.